src/main/java/org/jetbrains/bsp/bazel/server/bazel/data/BazelProcess.java

```diff
@@ -0,0 +1,60 @@
+package org.jetbrains.bsp.bazel.server.bazel.data;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+import org.jetbrains.bsp.bazel.server.bazel.utils.BazelStreamReader;
+import org.jetbrains.bsp.bazel.server.loggers.BuildClientLogger;
+
+public class BazelProcess {
+
+  private static final int OK_EXIT_CODE = 0;
+
+  private final Process process;
+  private final Optional<BuildClientLogger> buildClientLogger;
+
+  public BazelProcess(Process process, Optional<BuildClientLogger> buildClientLogger) {
+    this.process = process;
+    this.buildClientLogger = buildClientLogger;
+  }
+
+  public BazelProcessResult waitAndGetResult() {
+    try {
+      int exitCode = process.waitFor();
+      BazelProcessResult bazelProcessResult =
+          new BazelProcessResult(process.getInputStream(), process.getErrorStream(), exitCode);
+
+      logBazelMessage(bazelProcessResult, exitCode);
+
+      return bazelProcessResult;
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void logBazelMessage(BazelProcessResult bazelProcessResult, int exitCode) {
+    if (exitCode == OK_EXIT_CODE) {
+      logBazelMessage(bazelProcessResult);
+    } else {
+      logBazelError(bazelProcessResult);
+    }
+  }
+
+  private void logBazelMessage(BazelProcessResult bazelProcessResult) {
+    String message = bazelProcessResult.getJoinedStderr();
```

Comment:
is logging stderr here correct? if yes, maybe a comment clarifying that would be nice

Reply:
well, it is correct because it's copied, but why I don't know. Probably I will fix it somehow in the following PR trying to remove redundant empty lines in those logs